### ISSUE TITLE
Extend 'Rename Symbol' feature to support renaming macros

### ISSUE THREAD

**Issue by programminghoch10** [microsoft/vscode-cpptools#6194]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10
- VS Code Version: 1.49.1
- C/C++ Extension Version: v1.01
- Other extensions you installed (and if the issue persists after disabling them): PlatformIO, Does persist.
- Does this issue involve using SSH remote to run the extension on a remote machine?: No

Project size is irrelevant, happens also in one file as seen below.

**Steps to reproduce**

```cpp
#define TEST_CONSTANT "string"

void main() {
  std::cout << TEST_CONSTANT;
}
```
1. Try to rename the TEST_CONSTANT using F2 (Rename Symbol).
2. Notice it only renames usages, but not the define itself.

**Expected behavior**

It should also rename the constant itself.

**Comment by Colengms** [comment]
This feature is currently specific to semantic symbols.  For example, if you have functions named foo() in multiple classes, and select just one class's instance of foo(), it should not rename foo() in unrelated classes.  I don't believe it's intended to perform the equivalent of a Replace All, which can be done using the Search pane.  We can consider this a request to extend the feature to macro names, if possible.  We may be able to ensure that only sources leveraging the same #define statement (and not another unrelated #define of the same name) are modified.

**Comment by dzid26** [comment]
It's been couple years, can we start with just _Replacing All_ for macros? If I understand correctly, there shouldn't be double #defines anyway in the project. 

Right now it behaves so dodgy - I hover on the macro definition; I press F2 expecting it will behave just like variables; I put a new name; I am happy because I see all the usages were renamed;  **nope!** - the one that I just "renamed" didn't get renamed. You've got me again.

**Comment by iyadahmed** [comment]
Still not fixed :/

**Comment by costyn** [comment]
Sad that this bug is still not fixed :( But at least I know I'm not doing it wrong.

**Comment by xenkuo** [comment]
> This feature is currently specific to semantic symbols. For example, if you have functions named foo() in multiple classes, and select just one class's instance of foo(), it should not rename foo() in unrelated classes. I don't believe it's intended to perform the equivalent of a Replace All, which can be done using the Search pane. We can consider this a request to extend the feature to macro names, if possible. We may be able to ensure that only sources leveraging the same #define statement (and not another unrelated #define of the same name) are modified.

It still can't work.

BTW, the macro in language C is NOT considered as a semantic symbol? The "Go To Definition" can support it, the Outline panel can support it as well. 

To be honest, I'm not really understand the exact definition of the "semantic symbol" here..

**Comment by dzid26** [comment]
It _can_ be implemented the same as "Replace All". Just let user confirm with checkboxes which definitions they don't want to rename. 

At this point, instead of making it perfect, just make it work :)

**Comment by Dromantor** [comment]
I really don't get how such simple things can be broken for ages, while any IDE out there handles this properly from the start. (And no, "this is not an IDE, it's an editor, you are using it wrong" is not a catch-all excuse for things remaining half-broken. Other editors with C(++) support don't fail at this either.)

**Comment by Devin-goBILDA** [comment]
Yeah this issue has been plaguing me for a few years, would love to see some headway made on this!

**Comment by sorin0101** [comment]
So, after 5 years this is still not fixed. I'm starting to think that they do this intentionally, to push people to Visual Studio...

**Comment by sean-mcmanus** [comment]
@sorin0101 It was not "intentional" -- the find all references code was skipping macros for some reason, so when I change the code to stop skipping macros then rename works...I have to look into if there are any cases where there are problems with that change though. This should have been fixed 5 years ago (seems like it just fell through the cracks -- thanks for complaining it about it, otherwise we wouldn't have noticed).

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/v1.29.0 (pre-release)
